plugin/pkg/global/resourcereferencemanager/admission.go

```diff
@@ -319,6 +323,98 @@ func (r *ReferenceManager) Admit(ctx context.Context, a admission.Attributes, o
+			removedKubernetesVersions := sets.StringKeySet(helper.GetRemovedVersions(oldCloudProfile.Spec.Kubernetes.Versions, cloudProfile.Spec.Kubernetes.Versions))
+
+			// getting Machine image versions that have been removed from the CloudProfile
+			removedMachineImageVersions := map[string]sets.String{}
+			for _, oldImage := range oldCloudProfile.Spec.MachineImages {
+				imageFound := false
+				for _, newImage := range cloudProfile.Spec.MachineImages {
+					if oldImage.Name == newImage.Name {
+						imageFound = true
+						removedMachineImageVersions[oldImage.Name] = sets.StringKeySet(helper.GetRemovedVersions(oldImage.Versions, newImage.Versions))
+					}
+				}
+				if !imageFound {
+					for _, version := range oldImage.Versions {
+						if removedMachineImageVersions[oldImage.Name] == nil {
+							removedMachineImageVersions[oldImage.Name] = sets.NewString()
+						}
+						removedMachineImageVersions[oldImage.Name] = removedMachineImageVersions[oldImage.Name].Insert(version.Version)
+					}
+				}
+			}
+
+			if len(removedKubernetesVersions) > 0 || len(removedMachineImageVersions) > 0 {
+				shootList, err1 := r.shootLister.List(labels.Everything())
+				if err1 != nil {
+					return apierrors.NewBadRequest(fmt.Sprintf("could not list shoots to verify Kubernetes and Machine image version: %s", err1.Error()))
+				}
+
+				var (
+					channel = make(chan error)
+					wg      sync.WaitGroup
+				)
+				wg.Add(len(shootList))
+
+				for _, s := range shootList {
+					shoot := s.DeepCopy()
+					go func(ctx context.Context, shoot *core.Shoot, cloudProfile *core.CloudProfile, removedKubernetesVersions sets.String, removedMachineImageVersions map[string]sets.String) {
+						defer wg.Done()
+
+						if shoot.Spec.CloudProfileName != cloudProfile.Name {
+							channel <- nil
+							return
+						}
+
+						if removedKubernetesVersions.Has(shoot.Spec.Kubernetes.Version) {
+							channel <- fmt.Errorf("unable to delete Kubernetes version %q from CloudProfile %q  - version is still in use by shoot %q/%q", shoot.Spec.Kubernetes.Version, shoot.Spec.CloudProfileName, shoot.Name, shoot.Namespace)
+						}
+						for _, worker := range shoot.Spec.Provider.Workers {
+							if worker.Machine.Image == nil {
+								channel <- nil
+								return
+							}
+							// happens if Shoot already runs a version that does not exist in the CloudProfile - in this case: ignore
+							if _, ok := removedMachineImageVersions[worker.Machine.Image.Name]; !ok {
+								channel <- nil
+								return
+							}
+							if removedMachineImageVersions[worker.Machine.Image.Name].Has(worker.Machine.Image.Version) {
+								channel <- fmt.Errorf("unable to delete Machine image version %q/%q from CloudProfile %q - version is still in use by shoot %q/%q by worker %q", worker.Machine.Image.Name, worker.Machine.Image.Version, shoot.Spec.CloudProfileName, shoot.Name, shoot.Namespace, worker.Name)
```

Comment:
Using `%q/%q` looks kind of weird to me (e.g. `"ubuntu"/"1.17.1"`). WDYT?
```suggestion
								channel <- fmt.Errorf("unable to delete Machine image version '%s/%s' from CloudProfile %q - version is still in use by shoot '%s/%s' by worker %q", worker.Machine.Image.Name, worker.Machine.Image.Version, shoot.Spec.CloudProfileName, shoot.Namespace, shoot.Name, worker.Name)
```